### PR TITLE
Feature/setup plugin

### PR DESCRIPTION
**Pull request by b97pla**
- Initial setup as a plugin
- Moved code from main TACA repository

### CODE REVIEW

**Comment by b97pla** [comment]
@guillermo-carrasco would be great if you could review and give feedback. I've verified that the plugin is picked up and can be used together with TACA.

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    version=__version__,
+    description='Plugin for the TACA tool, bringing functionality for the '\
+        'ngi_pipeline to the table',
+    long_description='This package is a plugin that adds a set of subcommands '\
+        'to the TACA tool. The subcommands are useful when interacting with '\
+        'the ngi_pipeline. The TACA tool and the ngi_pipeline are used in the '\
+        'day-to-day tasks of bioinformaticians in National Genomics '\
+        'Infrastructure in Sweden.',
+    keywords='bioinformatics',
+    author='Guillermo Carrasco, Pontus Larsson',
```
Comment (guillermo-carrasco) [inline setup.py]: oh, did I do anything here? :-P

**Comment by guillermo-carrasco** [comment]
Looks great! And if you've tested that it is picked up and works, that should be enough. Just some questions:

* I guess you have to do the proper changes in `TACA`, like removing the `deliver` functionality. Have you done them and not pushed them? In that case I guess you've tested all of this in a fresh virtual environment?
* Is there any new code besides `setup.py` and travis files that I should check? I guess that it is mainly code from the already existing `deliver` funcitonality
* As this plugin is inherently dependent on `TACA`, wouldn't be better to use `TACA` logging module instead of writing a new one for the plugin?

Thanks Pontus, awesome!

**Comment by b97pla** [comment]
- I have, but I have only pushed it to [my fork](https://github.com/b97pla/TACA/tree/feature/deliver_plugin) and yes, tested in a fresh virtual environment.
- Now that you mention it, I did include some unmerged changes for caching checksums and a test related to that. It's probably easier for you to review if I stash those changes and only focus on the actual plugin to begin with. I'll close this PR and send a new, cleaner one.
- Indeed, I only made a small change to include the version of the code in the log messages but I could PR that to TACA if you think it's a good idea.

**Comment by guillermo-carrasco** [comment]
Ahh nice ^^! Yes please, I think that reusing TACA mail loggin module is a good idea :)

Tack tack!

**Pull request closed without merging**